elements/lisk-tree/src/sparse_merkle_tree/utils.ts

```diff
@@ -229,8 +231,31 @@ export const binaryStringToBuffer = (str: string) => {
 	return buf;
 };
 
-export const leafHash = (key: Buffer, value: Buffer): Buffer =>
-	hash(Buffer.concat([LEAF_HASH_PREFIX, key, value]));
+export const parseLeaf = (data: Buffer, keyLength: number): { key: Buffer; value: Buffer } => {
+	// Get the key of keyLength size
+	const key = data.slice(1, keyLength + 1);
+	// Get data
+	const value = data.slice(keyLength + 1, data.length);
+
+	return {
+		key,
+		value,
+	};
+};
+export const parseBranch = (data: Buffer): { leftHash: Buffer; rightHash: Buffer } => {
+	// Get left hash
+	const leftHash = data.slice(-2 * NODE_HASH_SIZE, -1 * NODE_HASH_SIZE);
+	// Get right hash
+	const rightHash = data.slice(-1 * NODE_HASH_SIZE);
+
+	return {
+		leftHash,
+		rightHash,
+	};
+};
+
+export const leafDigest = (key: Buffer, value: Buffer): Buffer =>
+	Buffer.concat([LEAF_HASH_PREFIX, key, value]);
 
-export const branchHash = (leftHash: Buffer, rightHash: Buffer): Buffer =>
-	hash(Buffer.concat([BRANCH_HASH_PREFIX, leftHash, rightHash]));
+export const branchDigest = (leftHash: Buffer, rightHash: Buffer): Buffer =>
+	Buffer.concat([BRANCH_HASH_PREFIX, leftHash, rightHash]);
```

Comment:
I don't think these functions names are conceptually correct here. We are not creating any digest rather creating message itself to be digested. Would rather suggest to use `branchMessage` or `branchData` or `calcBranchHashMessage` 

> A message digest is a cryptographic hash function containing a string of digits created by a one-way hashing formula.